src/data_projection_routines.f90

```diff
@@ -3608,5 +3656,55 @@ END SUBROUTINE DATA_PROJECTION_RESULT_XI_SET
   !================================================================================================================================
   !
 
+  !>Gets the projection vector for a data point identified by a given global number.
+  SUBROUTINE DATA_PROJECTION_RESULT_PROJECTION_VECTOR_GET(DATA_PROJECTION,DATA_POINT_USER_NUMBER,PROJECTION_VECTOR,ERR,ERROR,*)
+
+    !Argument variables
+    TYPE(DATA_PROJECTION_TYPE), POINTER :: DATA_PROJECTION !<A pointer to the data projection for which projection result is stored
+    INTEGER(INTG), INTENT(IN) :: DATA_POINT_USER_NUMBER !<The Data projection user number to get the projection xi for
+    REAL(DP), INTENT(OUT) :: PROJECTION_VECTOR(:) !<On exit, the projection vector of the specified global data point
+    INTEGER(INTG), INTENT(OUT) :: ERR !<The error code
+    TYPE(VARYING_STRING), INTENT(OUT) :: ERROR !<The error string
+    !Local Variables
+    INTEGER(INTG) :: DATA_POINT_GLOBAL_NUMBER
+    
+    ENTERS("DATA_PROJECTION_RESULT_XI_GET",ERR,ERROR,*999)
+
+    IF(ASSOCIATED(DATA_PROJECTION)) THEN
+      IF(DATA_PROJECTION%DATA_PROJECTION_FINISHED) THEN
+        IF(DATA_PROJECTION%DATA_PROJECTION_PROJECTED) THEN
+          CALL DataProjection_DataPointGlobalNumberGet(DATA_PROJECTION,DATA_POINT_USER_NUMBER, &
+            & DATA_POINT_GLOBAL_NUMBER,ERR,ERROR,*999)
+          IF(SIZE(PROJECTION_VECTOR,1)==SIZE(DATA_PROJECTION%DATA_PROJECTION_RESULTS(DATA_POINT_GLOBAL_NUMBER)%PROJECTION_VECTORS, &
```

Comment:
The test should probably be that the size was >= as we only just want to ensure there is enough memory.